### DISCUSSION TITLE
Not enabled extensions using the installer and a volume on /usr/local/etc/php/conf.d/

### DISCUSSION THREAD

**Discussion by adil-azmy** [mlocati/docker-php-extension-installer#588] [Q&A] [answered]
I'll just leave this here, in case someone run into the same problem as i did.

Case : 
- Having a Docker volume mapped to `/usr/local/etc/php/conf.d/` (./php-conf-ini/:/usr/local/etc/php/conf.d) with some custom ini files to build into the container.
```
volumes:
    - ./php-conf-ini/:/usr/local/etc/php/conf.d
```
- Since the directory is mapped as a volume, each time you run the container, a layer is added taking only the local files into consideration by php, and ignoring the new ini files created inside the container, those activating the extensions (ending up on the layer under inside the docker container)

Solution : Dont map the volume and add the custom ini files by copy

Example
```
COPY \
    # Custom settings
    ./php-conf-ini/custom.ini \
    # Add more Files if needed (by line = easy to comment and disable)
    ./php-conf-ini/errors_disable.ini \
    ./php-conf-ini/opcache.ini \
    # then... the line of the destination
    /usr/local/etc/php/conf.d/
```

Now the ini files enabling the extensions will be all active on running php configuration.

hope it will help

**Comment by mlocati** [comment, marked as answer]
Yep, that's the standard behavior for docker images: you shouldn't map a folder that may break the state (in this specific case: the PHP extensions enabled in INI the files in the `"$PHP_INI_DIR/conf.d"` directory should refer to existing .so files in the `"$(php-config  --extension-dir)"` directory.
